### ISSUE TITLE
0.20.0rc1: Broken -logthreadnames on gitian builds

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#18652]
On `0.20.0rc1`, or on just a gitian build against master (4702cadca94520a36bfe979c85750563c30f7c85), the `debug.log` contains empty thread names:

```
2020-04-15T12:37:41Z [] Bitcoin Core version v0.20.0rc1 (release build)
...
2020-04-15T12:37:41Z [] Config file arg: logthreadnames="1"
...
```

Tested on Linux Mint 19.3 (x86) and ODROID-HC1 (ARM, bionic).

**Comment by fanquake** [comment]
I've just tested the macOS 0.20.0rc1 bitcoind, and threadnames are working:
```bash
2020-04-15T12:53:46.857310Z [init] Bitcoin Core version v0.20.0rc1 (release build)
2020-04-15T12:53:46.857350Z [init] Assuming ancestors of block 0000000000000000000f2adce67e49b0b6bdeb9de8b7c3d7e93b21e7fc1e819d have valid signatures.
2020-04-15T12:53:46.857356Z [init] Setting nMinimumChainWork=00000000000000000000000000000000000000000e1ab5ec9348e9f4b8eb8154
...
2020-04-15T12:53:47.312306Z [shutoff] zmq: Shutdown notifier pubrawtx at tcp://127.0.0.1:28332
2020-04-15T12:53:47.312327Z [shutoff] zmq: Close socket at address tcp://127.0.0.1:28332
2020-04-15T12:53:47.313334Z [shutoff] Shutdown: done
```

Can you provide more information.

**Comment by hebasto** [comment]
I observed this bug on Linux systems (have not test other platforms yet).

The first system:
```
$ uname -sm
Linux x86_64
```

The second one:
```
$ uname -sm
Linux armv7l
```

Tested binaries are from https://bitcoincore.org/bin/bitcoin-core-0.20.0/test.rc1/

**Comment by hebasto** [comment]
The same issue with official downloads `0.19.0.1` and `0.19.1` ((

**Comment by hebasto** [comment]
The bug is caused by [`-fvisibility=hidden`](https://gcc.gnu.org/onlinedocs/gcc-7.5.0/gcc/Code-Gen-Options.html) compiler option after `configure --enable-reduce-exports`.

Not sure which solution would be correct...

**Comment by hebasto** [comment]
The observed behavior was introduced in #16059.

According to that pr, there are some ways to make `-logthreadnames` work on gitian builds:
- force `--enable-threadlocal`
or
- drop `--enable-glibc-back-compat`

What could you suggest, @dongcarl ?

**Comment by fanquake** [comment]
So to clarify, nothing is actually broken, we [explicitly don't use thread_local](https://github.com/bitcoin/bitcoin/blob/master/configure.ac#L864) when `--enable-glibc-back-compat` is passed, which is the case for [gitian builds](https://github.com/bitcoin/bitcoin/blob/master/contrib/gitian-descriptors/gitian-linux.yml#L44). Hence why they are work for macOS, but not for linux.

**Comment by hebasto** [comment]
> Hence why they are work for macOS, but not for linux.

Because [macOS gitian build](https://github.com/bitcoin/bitcoin/blob/544709763e1f45148d1926831e07ff03487673ee/contrib/gitian-descriptors/gitian-osx.yml#L41) does not use `--enable-glibc-back-compat`.

**Comment by fanquake** [comment]
> Because macOS gitian build does not use --enable-glibc-back-compat.

Thanks. That wasn't a question, just a typo. s/are//

**Comment by hebasto** [comment]
@fanquake 
> although obviously we should do something about having a runtime option available that will never actually do anything.

See: #18665.

**Comment by laanwj** [comment]
> So to clarify, nothing is actually broken, we explicitly don't use thread_local when --enable-glibc-back-compat is passed,

I don't know when thread-local support was added. Is this still necessary after the minimum glibc bump to 2.17?

**Comment by jasdeep9899** [comment]
I am doing a project and for that I need minimum 150 btc testnet coins... Can anyone help me please... My btc testnet wallet address is

mma9KKAAu32M5fJMchMoZJNB1xBJxJ5Z8c

**Comment by hebasto** [comment]
I think this issue is fixed by #18665.
